docs/sdg/sdg-flow-yaml.md

```diff
@@ -0,0 +1,157 @@
+
+### Prompt Config Files
+
+Every LLMBlock references a separate prompt config file, and presumably a custom pipeline will provide custom prompt configs too.
+
+These prompt config files are quite simple YAML files - they contain a single object with system, introduction, principles, examples, and generation keys. See e.g. src/instructlab/sdg/configs/skills/freeform_questions.yaml
+
+We will continue to use these config files unchanged, and custom files can be specified with an absolute path. Relative paths are assumed to be relative to the Python package e.g. `configs/skills/...`.
+
+### Future - Model Serving
+
+Custom pipelines may have more unique model serving requirements. Instead of serving just one model, we may need to launch the model server with a model and an additional model with adapter.
+
+Right now the sdg code takes two parameters for each block - the OpenAI client instance, and the model ID. This should be sufficient. vLLM, for example, can host both a model and a model+adapter under two different model IDs.
+
+In future, it may make sense to define some model serving config (e.g. which adapter to use for SDG) as part of a flow definition. We will address this in a future if enhancement, if needed.
+
+### CLI Integration
+
+As of the current version of `ilab`, it supports `simple` and `full` as parameters to `--pipeline` to select one of the two types of built-in pipelines included in the library. Once we have support for loading custom pipelines, we need a way for these to be specified with the CLI. Here is a proposed CLI UX for this:
+
+> `ilab data generate`
+
+Use the default pipeline, `simple`.
+
+> `ilab data generate --pipeline full`
+
+Use the built-in `full` pipeline.
+
+> `ilab data generate --pipeline full --pipeline-extend path/to/config.yaml`
+
+Use the `full` pipeline, but extend it with the pipeline defined in the provided config file.
+
+> `ilab data generate --pipeline none --pipeline-extend path/to/other_config.yaml`
+
+Do not use a built-in pipeline. Use only the pipeline in the custom configuration.
+
+### Future CLI Improvements
+
+A possible improvement would be to have a well-defined place on the filesystem where custom pipeline configs can be automatically loaded and included as options to the `--pipeline` parameter.
+
+For example, if the config format included new parameters, `name: full-extended` and `extends: full`, and the CLI discovered and loaded it automatically, we could support `--pipeline full-extended` without needing the additional `--pipeline-extend` option.
+
+`/usr/share/instructlab/sdg/` is a proposed location for this as a place for a distribution of InstructLab to include pre-defined custom pipelines, at least for Linux. See the [Filesystem Hierarchy Standard](https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch04s11.html) for more details on why this path is appropriate for this use-case.
```

Comment:
If we have a location with pipeline examples then a nice to have would be to have a `ilab date generate --list-pipelines` perhaps? Not necessarily in the scope of this design but food for thought.

Reply:
Added